waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/SessionFactorySupplier.java

```diff
@@ -50,6 +52,25 @@ public SessionFactorySupplier(int sshPort, String knownHosts, List<String> ident
     this.knownHosts = knownHosts;
     this.identityKeys = ImmutableList.copyOf(identityKeys);
     this.sshTimeout = sshTimeout;
+    this.strictHostKeyChecking = "yes";
```

Comment:
calll the new constructor here like `this(,,,,,,"yes")` and do all the argument checking in one constructor